dot/parachain/collator-protocol/validator_side.go

```diff
@@ -397,6 +767,48 @@ func (f fetchedCollationInfo) String() string {
 		f.relayParent.String(), f.paraID, f.candidateHash.Value.String(), f.collatorID)
 }
 
+func (f *fetchedCollationInfo) FromString(str string) error {
```

Comment:
Kinda makes more sense for this to be a constructor.  Simplifies the calling code too.  Usually the `From` trait isn't doesn't reference `self`.